DOCS-1600 Azure App Service Extension

ruthnaebeck opened the pull request:
<!-- *Note: Please remember to review the Datadog Documentation [Contribution Guidelines](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md) if you have not yet done so.* -->

### What does this PR do?
Update Azure App Service Extension for GA

### Motivation
Request from PM

### Preview
https://docs-staging.datadoghq.com/ruth/docs-1600/serverless/azure_app_services/

### Additional Notes
N/A

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

steveharrington1 commented:
@ruthnaebeck a couple of thoughts:

1. "Application logging from Azure Web Apps" and "Requirements" should probably both be their own sections, not subsections of Setup. I suggest we just promote each part of "setup" to be their own section (and drop the "setup" grouping altogether).

2. I think we should add a link to Microsoft docs- in the first sentence let's make "Azure App Service" a link to https://docs.microsoft.com/en-us/azure/app-service/

ruthnaebeck commented:
> 1. "Application logging from Azure Web Apps" and "Requirements" should probably both be their own sections, not subsections of Setup. I suggest we just promote each part of "setup" to be their own section (and drop the "setup" grouping altogether).

The Setup section is standard on the documentation site. It follows the pattern:
Setup
- Requirements (or Prerequisites)
- Installation
- Configuration

Moved `Application logging from Azure Web Apps` to its own section.

> 2. I think we should add a link to Microsoft docs- in the first sentence let's make "Azure App Service" a link to https://docs.microsoft.com/en-us/azure/app-service/

updated

> 3. In installation step 5, the first line should read: Set ‘DD_SITE’ to the appropriate Datadog domain:
>    -datadoghq.com (default)
>    -datadoghq.eu
>    -us3.datadoghq.com
>    -ddog-gov.com

We use a "shortcode" (`{{< region-param key="dd_site" code="true" >}}`) instead of listing these out. Each site is / will be available based on the customer's selection on the site selector (top right). Right now you can toggle to EU. When subsequent sites are releases, customers who toggle to a different site will see their parameter.

steveharrington1 commented:
@ruthnaebeck Thanks for the note. The edits look good to me!

steveharrington1 commented:
Hi @ruthnaebeck As we spoke about yesterday, we should push this to Feb 1st (tentative). I also wanted to add a couple additional items:

**1. In requirements section, lets make this is list, and then add a 4th item:**

"
1. If you haven't already...
2. The Datadog .NET extension...
3. Datadog recommends...
[NEW} 4. Datadog automatic instrumentation relies on the .NET CLR Profiling API. This API allows only one subscriber (for example, APM). To ensure maximum visibility, run only one APM solution within your application environment."

**2. In the troubleshooting section let's update the beginning of the section:**

In many cases, issues can be resolved by doing a full **Stop** and **Start** of your web application. If that doesn't work, try uninstalling and re-installing the extension.

Still Having trouble? Contact Datadog support.

To expedite the process...

steveharrington1 commented on content/en/serverless/azure_app_services.md:
````diff
@@ -11,186 +11,126 @@ further_reading:
-3. Reinstall the Datadog extension.
-4. Restart your application.
+```csharp
+using (var statsService = new DogStatsdService())
````
@ruthnaebeck we have an updated example code block we think will make this more clear + help prevent tickets from a transient error we have seen in some cases:

try
{
    // Configure your DogStatsd client and configure any tags
    DogStatsd.Configure(new StatsdConfig() { ConstantTags = new[] { "app:sample.mvc.aspnetcore" } });
}
catch (Exception ex)
{
    // An exception will be thrown by the Configure call if the necessary environment variables are not present.
    // These environment variables are present in Azure App Services
    // But will need to be set in order to test your custom metrics: DD_API_KEY:{api_key}, DD_AGENT_HOST:localhost 
    // Ignore or log the exception as it suits you
    Console.WriteLine(ex);
}
// Send a metric
DogStatsd.Increment("sample.startup");

steveharrington1 commented on content/en/serverless/azure_app_services.md:
```diff
@@ -11,186 +11,126 @@ further_reading:
+
+3. Datadog recommends doing regular updates to the latest version of the extension to ensure optimal performance, stability, and availability of features.
+
+4. Datadog automatic instrumentation relies on the .NET CLR Profiling API. This API allows only one subscriber (for example, APM). To ensure maximum visibility, run only one APM solution within your application environment.
```
@ruthnaebeck Let's change this from the 4th bullet item to instead be a note at the end, ie:

**Note**: Datadog automatic instrumentation...

steveharrington1 commented on content/en/serverless/azure_app_services.md:
```diff
@@ -11,186 +11,135 @@ further_reading:
+    - .NET Core 3.0 (Microsoft support ended 2020-03-03)
+    - .NET Core 3.1
+
+3. Datadog recommends doing regular updates to the latest version of the extension to ensure optimal performance, stability, and availability of features.
```
Please add a note about requiring a restart.

3. Datadog recommends doing regular updates to the latest version of the extension to ensure optimal performance, stability, and availability of features. Not that both the initial install and subsequent updates require a restart of your web app.

zacharycmontoya commented on content/en/serverless/azure_app_services.md:
```diff
@@ -10,187 +10,142 @@ further_reading:
-- .NET Core 2.2 (Microsoft support ended 2019-12-23)
-- .NET Core 3.0 (Microsoft support ended 2020-03-03)
-- .NET Core 3.1
+    - .NET Framework 4.7 and later
```
```suggestion
    - .NET Framework 4.5 and later
```

The required framework of .NET Framework 4.5 and later will be consistent with our .NET Tracer requirement docs.

zacharycmontoya commented on content/en/serverless/azure_app_services.md:
```diff
@@ -10,187 +10,142 @@ further_reading:
+    - .NET Core 2.1
+    - .NET Core 2.2 (Microsoft support ended 2019-12-23)
+    - .NET Core 3.0 (Microsoft support ended 2020-03-03)
+    - .NET Core 3.1
```
```suggestion
    - .NET Core 3.1
    - .NET 5
```

We're recently added support in the .NET Tracer for .NET 5, the newest version of the ".NET Core" product

apigirl commented on content/en/serverless/azure_app_services.md:
```diff
@@ -10,187 +10,142 @@ further_reading:
   text: "Azure App Service Environment"
+- link: "https://www.datadoghq.com/pricing/?product=apm#apm-what-is-considered-as-a-host-for-azure-app-services"
+  tag: "Pricing"
+  text: "Datadog Pricing page"
```
```suggestion
  text: "Azure App Service APM Pricing"
```
